zcash_client_backend/src/data_api/testing/pool.rs

```diff
@@ -3245,3 +3246,86 @@ pub fn pczt_single_step<P0: ShieldedPoolTester, P1: ShieldedPoolTester, DSF>(
     let (h, _) = st.generate_next_block_including(txid);
     st.scan_cached_blocks(h, 1);
 }
+
+#[cfg(all(feature = "transparent-inputs"))]
+pub fn wallet_recovery_computes_fees<T: ShieldedPoolTester>(
+    ds_factory: impl DataStoreFactory,
+    cache: impl TestCache,
+) {
+    use secrecy::ExposeSecret;
+
+    let mut st = TestBuilder::new()
+        .with_data_store_factory(ds_factory)
+        .with_block_cache(cache)
+        .with_account_from_sapling_activation(BlockHash([0; 32]))
+        .build();
+
+    let seed = Secret::new(st.test_seed().unwrap().expose_secret().clone());
+    let source_account = st.test_account().cloned().unwrap();
+    let (dest_account_id, dest_usk) = st
+        .wallet_mut()
+        .create_account("dest", &seed, source_account.birthday(), None)
+        .unwrap();
+
+    let from = T::test_account_fvk(&st);
+    let (to, _) = dest_usk.default_transparent_address();
+
+    // Get some funds in the source account
+    let note_value = Zatoshis::const_from_u64(350000);
+    st.generate_next_block(&from, AddressType::DefaultExternal, note_value);
+    st.generate_next_block(&from, AddressType::DefaultExternal, note_value);
+    st.scan_cached_blocks(source_account.birthday().height(), 2);
+
+    // Create a sequence of three transactions: two transactions sending from the source account to
+    // a transparent address in the destination account, and then a shielding transaction in the
+    // destination account that pays appropriate fees.
```

Comment:
This comment is incorrect in the commit it is added in (5e41581da74df4d51798c3e5ddcdf180782fd2b1); only the first two transactions are being made. The third (shielding) transaction is added to the test in 90527446eec76bf8c90e778cdf8891279a473aa5.